.github/workflows/build.yml

```diff
@@ -0,0 +1,70 @@
+name: Build Firedrake
+
+on:
+  # Push to master or PR
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  build:
+    name: "Build Firedrake"  
+    # The type of runner that the job will run on
+    runs-on: self-hosted
+    # The docker container to use.
+    container:
+      image: firedrakeproject/firedrake-env:latest
+    strategy:
+      # Don't immediately kill real if complex fails and vice versa.
+      fail-fast: false
+      matrix:
+        include:
+         - scalar-type: real
+           complex: ""
+         - scalar-type: complex
+           complex: --complex
+    env:
+      FIREDRAKE_CI_TESTS: 1
+      PETSC_CONFIGURE_OPTIONS: --with-make-np=12 --download-mpich-device=ch3:sock
+      OMP_NUM_THREADS: 1
+      OPENBLAS_NUM_THREADS: 1
```

Comment:
@JDBetteridge I think we wanted to set these in the docker files, if that is possible?

Reply:
I think that's a separate issue. All we're doing right now is duplicating what Jenkins does.

Reply:
Sure.

Reply:
Having had a look, it doesn't seem possible to set the runtime environment variables when building the docker container. You can set them from the command line at runtime, or you can create a list of default environment variables in a file named `.env` in the same directory as the docker container.

References
build time: https://docs.docker.com/engine/reference/commandline/build/
runtime: https://docs.docker.com/compose/environment-variables/

Reply:
This seems far complicated than I meant:
```
FROM ubuntu:20.04
ENV OMP_NUM_THREADS 1
```

If I build that container and then run it, `OMP_NUM_THREADS` is set to 1 inside the container. This is all we want.